wire-protoc-integration/src/main/java/com/squareup/wire/protocwire/ProtocContext.kt

```diff
@@ -0,0 +1,47 @@
+package com.squareup.wire.protocwire
+
+import com.squareup.wire.schema.ClaimedDefinitions
+import com.squareup.wire.schema.ClaimedPaths
+import com.squareup.wire.schema.EmittingRules
+import com.squareup.wire.schema.ErrorCollector
+import com.squareup.wire.schema.Location
+import com.squareup.wire.schema.Profile
+import com.squareup.wire.schema.ProfileLoader
+import com.squareup.wire.schema.ProtoFile
+import com.squareup.wire.schema.Schema
+import com.squareup.wire.schema.SchemaContext
+import com.squareup.wire.schema.SchemaHandler
+import okio.Path
+
+data class ProtocContext(
+    private val response: Plugin.Response,
+    override val sourcePathPaths: Set<String>
+) : SchemaContext {
+  override val errorCollector: ErrorCollector = ErrorCollector()
+  override val emittingRules: EmittingRules = EmittingRules()
+  override val claimedDefinitions: ClaimedDefinitions? = null
+  override val claimedPaths: ClaimedPaths = ClaimedPaths()
+  override val module: SchemaHandler.Module? = null
+  override val profileLoader: ProfileLoader = object : ProfileLoader {
+    private val profile = Profile()
+    override fun loadProfile(name: String, schema: Schema): Profile {
+      return profile
+    }
+  }
+
+  override fun logArtifactHandled(qualifiedName: String, targetName: String) {
+    // no-op
+  }
+
+  override fun inSourcePath(protoFile: ProtoFile): Boolean {
+    return inSourcePath(protoFile.location)
+  }
+
+  override fun inSourcePath(location: Location): Boolean {
+    return location.path in sourcePathPaths
```

Comment:
The other impl (`FileSystemContext`) instead has this:
```kotlin
return sourcePathPaths == null || location.path in sourcePathPaths
```
Was the change here intentional? Does your version here blow up if `sourcePathPaths == null`? Or does it now return false when the other version would have returned true?

Reply:
Chatted offline -- This is no longer relevant since we're removing this class completely.